### DISCUSSION TITLE
Blockifying the Add to Cart with Options block: some decisions and next steps

### DISCUSSION THREAD

**Discussion by Aljullu** [woocommerce/woocommerce#50484] [WooCommerce, Blocks, and Site Editing]
* This is a follow-up of https://github.com/woocommerce/woocommerce/discussions/49799.

Over the past few weeks, @samueljseay and I have been exploring how the Add to Cart with Options block could be blockified. While we don't have immediate plans to implement this, we wanted to share our findings and thoughts on how it could work.

In other words, this discussion is not a commitment that we are going to implement this soon or in this specific way, but rather some early ideas of how it could be done. This way, WooCommerce developers and the community in general can raise a flag if something doesn't look good.

### Inner blocks

So far, we have identified the following inner blocks that will need to be created:

* Quantity Selector
* Add to Cart Button / External Product Link
* Low Stock Notice / Out of Stock Notice / On Backorder Notice
* Variation Selector
* Grouped Product Selector

Something that will need to be decided in a later stage is whether some of the blocks get grouped or not. Ie: Add to Cart Button and External Product Link could be the same block or two different blocks. Same for the stock notice blocks.

### Handling different blocks per product type

> [!NOTE]
>
> You can read the full GitHub discussion here: https://github.com/woocommerce/woocommerce/discussions/49799.

Depending on the product type, certain blocks (e.g., Variation Selector or Grouped Product Selector) need to be conditionally rendered in the editor.

Several ideas were proposed on how to render different blocks in the editor depending on the product type. We received quite a lot of feedback from WooCommerce developers as well from the community (thanks for that!). As the next steps, we would like to explore the "template states" idea (https://github.com/WordPress/gutenberg/issues/64207), as it received interest from Gutenberg designers.

We consider this to be an upstream contribution that will require a big effort, and it might be worth exploring it in parallel to "block states" (https://github.com/WordPress/gutenberg/issues/57719). We expect that both "block states" and "template states" will work well if WooCommerce [shifts from product types to product traits](https://developer.woocommerce.com/2024/08/06/request-for-feedback-simplifying-woocommerce-products-a-shift-from-types-to-traits/).

### Add to Cart wrapper block

There are two ways we can blockify the Add to Cart Form:

* Creating the inner blocks and allowing them to be placed anywhere in the template.
* Creating the inner blocks and a wrapper (or container) block, so inner blocks can only be placed inside the wrapper block.

We haven't made a final decision, but we lean towards starting with option 2 because it offers several benefits:

1. It allows us to have a conceptual block which adds the `<form>` element, which is required for accessibility reasons.
2. In case we decide not to, or can't, implement "template states", having an Add to Cart wrapper block would allow us to switch between product types using "block states".
3. We believe it will make supporting Add to Cart forms from 3rd-party product types easier (more info about that below).

Sidenote: the stock notice blocks could possibly escape the restriction and be placed anywhere in the template.

### Extensibility on 3rd-party product types

> [!NOTE]
> 
> Small note: WC core comes with four product types (simple, variable, grouped and external). By "3rd-party product types", we refer to product types created by extensions, like a Composite Product.

Depending on the product type, WC core renders different templates for the Add to Cart Form:

https://github.com/woocommerce/woocommerce/blob/a579e866b7a490a902781fda32fa13edf921e31c/plugins/woocommerce/src/Blocks/BlockTypes/AddToCartForm.php#L83

Plugins that add new product types are expected to provide their own templates.

Based on this, we would continue rendering the PHP templates provided by plugins for product types not included in WC core. This means, the Add to Cart Form block would continue being a "black box" block. This ensures backwards compatibility.

Extensions would be able to update the Add to Cart Form of their product types when they are ready to do so, they could reuse some of the blocks WC core provides (see previous section: _Inner blocks_) or create their own.

We would need to find a way for the Add to Cart Form block to know what to render in the editor, either the "black box" or the blockified experience. We propose a solution for this in the section _Synced patterns_ below.

### Extensibility on WC core product types

For WC core product types (simple, variable, grouped and external), we would simply proceed with blockifying the form, following the list of blocks described in _Inner blocks_. Hopefully, we will have a way to switch between product types (see _Handling different blocks per product type_ above).

As we blockify the experience, we will migrate to the Interactivity API (from now on, iAPI). This poses a risk as some extensions might be using some PHP hooks to render additional form fields. In those cases:

* We will experiment reading those fields via the iAPI and handling the form submission.
* If not possible, we would need to conditionally enable/disable the iAPI depending on the usage of certain PHP hooks via extensions (see [example code](https://github.com/woocommerce/woocommerce/compare/try/conditional-add-to-cart-form-block?expand=1)).

Migrating to the iAPI would allow the long-awaited Add to Cart action without a page refresh.

Small note: _Simple_ and, specially, _External_ products are the simplest ones to blockify, so we could start with those two.

### Synced patterns

One idea to facilitate the transition to the blockified Add to Cart Form is to create synced patterns (aka template parts) for each product type, with each pattern representing the contents of the Add to Cart Form.

For each product type, WC could easily check if there is a synced pattern. If one exists, WC would load it instead of the "black box" block.

This would allow extensions which register custom product types to easily migrate to the blockified experience: if they don't provide the synced pattern, WC will continue loading the PHP template and the "black box" block. Once they provide the synced pattern, WC will load it instead of the PHP template.

While this is an idea we had, we still consider it's a bit too early to decide, and we would like to see how the UX of "template states" evolve in Gutenberg before committing to using synced patterns or not. The possible [shift to product traits](https://developer.woocommerce.com/2024/08/06/request-for-feedback-simplifying-woocommerce-products-a-shift-from-types-to-traits/) could also influence whether this is the best path forward.

## Feedback

Although we don't anticipate this being shipped in the short term, we wanted to share this exploration publicly to allow extension developers and the community to share their feedback.

**Comment by 79ho3ein** [comment]
Hi woocommerce team,

>So far, we have identified the following inner blocks that will need to be created:

>Quantity Selector
>Add to Cart Button / External Product Link
>Low Stock Notice / Out of Stock Notice / On Backorder Notice
>Variation Selector
>Grouped Product Selector

It is wonderful that China has a plan. But I want to add one thing to the new blocks you mentioned.

**Price block**

Now, on variable products, a new price will be displayed when the user selects variables. Therefore, the user sees two different prices. (This is something our customers complain about all the time. They say, why are two prices displayed on the product page and are you confusing us?)
What is very, very important here is that when the user selects variables, a new price is not displayed, but the current price is updated in Ajax.  

Below is a comparison of the current situation and the proposed situation.

**current :**

https://github.com/user-attachments/assets/6072f7ce-156a-4df8-8cc1-0d9f16f27aec


**Improved :**

https://github.com/user-attachments/assets/5835d415-e200-42af-a858-17bbcd4c5986


Thanks

**Reply by Aljullu** [reply]
Thanks for mentioning this, @79ho3ein! I don't think this is something that we will solve in the first iteration of blockifying the Add to Cart with Options block, as we will first focus on achieving feature parity. However, it's definitely something we can look into in the future, once we start working on improving the inner blocks, including the Product Price block. I see what you mean that having both prices at the same time in the screen might be confusing to shoppers.

**Comment by op6sie** [comment]
can it be changed to have all in one line (again dutch)  5,90 Beschikbaar via nabestelling, it is always on 2 lines.
button size is huge and the amount selection  is as big as the button aka huge
Can this be changed so that those basic things can be made smaller?

**Reply by Aljullu** [reply]
I would say so. The goal of blockifying the Add to Cart with Options bock is to make all these things easily customizable.

**Reply by ResponsiveFlow** [reply]
@Aljullu has any work begun? Any ETA?

**Reply by Aljullu** [reply]
No, we hadn't started with this project yet. It's something we know we want to work on soon, but didn't have the chance yet. So I can't give an ETA yet. :slightly_smiling_face: 

At the same time, @helgatheviking is doing [some explorations](https://woocommercecommunity.slack.com/archives/C03MCL9RTAR/p1729778820228059) related to this project right now.

**Reply by helgatheviking** [reply]
I'm only looking at quantity input and maybe add to cart button inner blocks at the moment. There's still a lot up in the air related to exactly how everything is going to be implemented, but those inner blocks feel like they should cross over to almost any implementation.

**Comment by Aljullu** [comment]
For everybody following this discussion: we started with this project. :slightly_smiling_face: 

We will work on it in different phases, and we are starting by creating a simple version of the inner blocks. After that, we will work on adding a compatibility layer, the Interactivity API and handling different product types in the editor.

We don't have an ETA and there is nothing ready to test yet. But if you are interested, I would recommend you to follow the ongoing progress in the epic issue (https://github.com/woocommerce/woocommerce/issues/53014) and the [Project Board](https://github.com/orgs/woocommerce/projects/284).

**Reply by 79ho3ein** [reply]
Thanks for the great news @Aljullu💪 Good luck.

**Comment by 79ho3ein** [comment]
Hi woocommerce team,

A question about this feature...

If we use the Variation Selector block, in the frontend for simple products, will this block be removed from the DOM?

**Reply by Aljullu** [reply]
Yes! :100: If we proceed with the "Synced patterns" approach, there will be an "Add to Cart" synced pattern for each product type, which will only contain the blocks required for that product type. At render time, the correct pattern will be rendered.

If, for some reason, the block is in the page in the wrong product type (ie: the user inserted it by copy & pasting the block HTML markup in the wrong template), the [block won't render anything](https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce/src/Blocks/BlockTypes/AddToCartWithOptionsVariationSelector.php#L319-L322).

**Comment by 79ho3ein** [comment]
Can the clear button also be a block in variable products?

Suppose the variable selection is in the form of button. Like [this](https://prnt.sc/RWC7aBdN6WzE). In this situation, there is no need for a clear button. Because by clicking on the desired color, it is selected, and after clicking again, it is deleted and acts like a clear button. So if the clear button is a block, we can delete it if needed.

**Reply by Aljullu** [reply]
Thanks for the feedback, @79ho3ein! We hadn't started to experiment with it yet, but I know design will look into it and using "pill shapes" for the options is something they are considering. I passed your feedback internally so they are aware of it. :+1:

**Reply by 79ho3ein** [reply]
Thank you @Aljullu

**Comment by 79ho3ein** [comment]
Hi @Aljullu, Good time!

There is a significant issue when selecting variables in products that I'm not sure is within the scope of your team, but since you are working on the new Add to Cart block, it seems like you will encounter it.

**The problem is that if the number of variables is more than 30, Ajax will have trouble updating the variables.** I've brought this up [here](https://github.com/woocommerce/woocommerce/issues/51556) before and it seems to be the default behavior. For performance reasons.

I may be wrong, but it seems logical if there is no limit in this case. Because performance is affected based on the number of variables. So if someone has 5 variables for their products, they will definitely not face performance problems. On the other hand, if the number of variables is large, it is naturally expected that performance will be affected.

WooCommerce's current approach to this is like assuming a site builder like Elementor says you are allowed to use a maximum of 40 elements per page, because you will encounter performance issues with more than that number. This doesn't seem logical.

In general, if WooCommerce removes the current limitation, will there be any problem? Because many users will definitely face this challenge. As many customers reported this to us and I spent hours searching for the reason.

thank you

**Reply by Aljullu** [reply]
Thanks for bringing it to our attention, that's a good point! As we are planning to redesign the variation selector, it will be a good moment to revisit the default value for `woocommerce_ajax_variation_threshold` or, even if the limit is needed at all. It's still a bit too early to make any promises, but I will keep it in mind.

**Reply by 79ho3ein** [reply]
Nice, thank you

**Comment by helgatheviking** [comment]
>As we are planning to redesign the variation selector,

Please say this means swatches 😅

**Reply by Aljullu** [reply]
> Please say this means swatches 😅

I hadn't seen the designs yet, but I think the idea is to have pill-shaped selectors instead of classic dropdowns. :slightly_smiling_face:

**Reply by sunyatasattva** [reply]
Also, in theory, whatever design we make, the mere fact of blockifying the entire section, would mean that it shouldn't be too hard for people to create their own block for their specific design need. At least I hope. Similarly to what we are doing with the Filters block.

**Reply by helgatheviking** [reply]
blocks and "shouldn't be too hard" are not phrases that go together for me :) but I do love getting rid of the dropdowns. I have always thought that could be better.

**Reply by helgatheviking** [reply]
I checked out the "pill" design and it's pretty cool!

**Comment by Joliamendi** [comment]
Hi woocommerce team,

Given the extensive updates on single products, is there a plan to display the sale bagde as a percentage? In variable products, the percentage value is updated via Ajax after each variable is selected.

### More efficient display of price and discount percentage on variable products in the loop

As you know, currently, the prices of variable products are displayed as ranges. And sale bade is displayed only as plain text (SALE). I think this could be much better and we could make some improvements here.

1- Choose which variable to display as the default for variable products. The variable with the lowest price or the variable with the highest price. This helps prevent variable product prices from being displayed as a range.

2- The previous case also applies to the discount percentage. Therefore, according to the selection we made in the previous step, the discount percentage will be displayed.

### What happens in single product of variable products?

In single product, the price and discount percentage display what we selected in the previous step. When the user selects the desired variable, the price and discount percentage are updated.

### How does this help?

Frankly, WooCommerce's current approach to displaying variable product prices (whether in loop or a single product) is outdated and not user-friendly at all. The user initially sees only one price in the single product. After selecting the variable, a new price appears and now two prices are displayed! This is very misleading for users. Because we have had several reports where users have complained about the way the price is displayed.

### An example for this

Let's take a look at the Amazon website. Let's take a look at Amazon's website. [In the product listing or loop](https://prnt.sc/8gcaSLGLrvNc), there is only one price displayed for the iPhone 16. This is the lowest price among the iPhone 16 variants that they have chosen to display as the default.
On the other hand, [on the product page](https://prnt.sc/p24oJzKtSkvM), after selecting each variable, the price and product details are updated.

I hope the explanation was useful.

thank you

**Reply by Aljullu** [reply]
Thanks for the feedback, @Joliamendi! Your points make a lot of sense to me, and I agree they would be nice features to add. In the past, some improvements have been discussed for the Product Sale Badge and Product Price blocks, and your comments would be good additions.

I think it's a bit out of the scope of this discussion, though, so I would suggest [opening new issues](https://github.com/woocommerce/woocommerce/issues/new/choose) for them so they can be properly triaged and prioritized by our team. Thanks in advance!

**Comment by tuanfront-end** [comment]
@Aljullu Hope you are well and wondering if there is any start yet? 🤔

**Reply by Aljullu** [reply]
Yes, thanks for asking, @tuanfront-end! Progress continues on this project, I just published an update here: https://github.com/woocommerce/woocommerce/issues/53014#issuecomment-2637680874.

**Comment by 79ho3ein** [comment]
Hi @Aljullu,

First of all, a special thank you for the efforts you are making to make WooCommerce stronger. The new updates say it all and it is truly appreciated.

I wanted to ask if there is currently any plan to implement the price update feature in variable products? By price update I mean something I mentioned [here](https://github.com/woocommerce/woocommerce/discussions/50484#discussioncomment-10946918) before.

Thank you

**Reply by Aljullu** [reply]
Thanks for the interest, @79ho3ein. Yes, we are working on it here: https://github.com/woocommerce/woocommerce/pull/58384. It's likely the behavior will be slightly different than in classic themes/templates, instead of displaying a new price for the variation, the existing product price block will be updated.

**Reply by 79ho3ein** [reply]
Wow, that's great. What version will the feature be released in? Is it planned?

**Reply by Joliamendi** [reply]
This is awesome. But one question. If we want to display the inventory of variables, after selecting the variables, the inventory quantity is also updated immediately like the price?

**Reply by Aljullu** [reply]
> Wow, that's great. What version will the feature be released in? Is it planned?

We are planning to release the block in beta in WooCommerce 10.0, which is the next release. There will still be a few things to flesh out, that's why it's going to be labelled as beta. But it should be ready for most use-cases and folks who want to test it. :slightly_smiling_face: 

> This is awesome. But one question. If we want to display the inventory of variables, after selecting the variables, the inventory quantity is also updated immediately like the price?

Yes, when a variation is selected several blocks will update:

- Product Price
- Product Description
- Stock Indicator
- Product SKU
- Image Gallery
- ...

We are starting with the Product Price because it's the most critical one (shoppers being able to know how much the variation costs!). But the plan is to extend the behavior to all other blocks that should render variation data. We have an issue for this here: https://github.com/woocommerce/woocommerce/issues/56870.

**Reply by helgatheviking** [reply]
So @Aljullu I assume there will be an action or a store change we can listen to in order to detect the variation change?

**Reply by Aljullu** [reply]
Yes, I don't know exactly how it will look for extensions, as we need to test it in real use cases before finalizing the API, but I imagine the active variation ID will be available via the _Single Product_ block store (or the global state when in the template). This way, extensions can listen to it and update accordingly.

But as mentioned, we are still figuring out the internal API for our blocks, so it's a bit too early to say how it will look for 3rd-party blocks. :slightly_smiling_face:

**Comment by 79ho3ein** [comment]
Hi WooCommerce team,

I'd like to report a UX inconsistency and suggest a possible improvement related to the "Add to Cart" button behavior when product stock is very limited.

### Current Behavior

When a product has only 1 unit in stock:

1. The user clicks “Add to cart”.
2. The page reloads, and the button remains active.
3. If the user clicks it again, WooCommerce shows an error like:

> "You cannot add that amount to the cart — we have 1 in stock and you already have 1 in your cart."

While technically correct, this behavior creates a confusing experience for the customer.

### Why This Is a UX Problem

- The customer sees the "Add to cart" button still active, which implies they can add more.
- Receiving an error after clicking again can cause confusion or frustration.
- It may seem like a bug or poor stock handling, especially for stores with limited items or flash sales.

### Suggested Improvement

After the last available unit of a product is added to the cart (before checkout), the system should:

- **Disable the "Add to cart" button**, OR
- **Change the button label** to something like:
  - “Added to cart”
  - “In your cart”
  - “Out of stock”

This logic could optionally be toggled via a filter or setting, in case some store owners prefer the current behavior.

### Notes

This enhancement would not only improve clarity but also reduce user error and support requests due to misunderstandings around stock limits.

Thanks for your time and for considering this suggestion!

**Reply by Aljullu** [reply]
Thanks for the suggestion, @79ho3ein! I agree the shopper experience could be improved in this case. In general, I don't think it's a good idea to disable buttons, because some times it might not be clear why they are disabled. But I agree we could add some UI hints that more products can't be added to cart without requiring the user to click on the button again. Maybe changing the cursor, the button opacity or the text, as you mention.

@79ho3ein would you be willing to [open a new issue](https://github.com/woocommerce/woocommerce/issues) for this so we can add it to our tasks and prioritize? :bow:

**Reply by 79ho3ein** [reply]
Hi again @Aljullu,

>I don't think it's a good idea to disable buttons

Actually, by disabling the button, I meant something like reducing the opacity and changing the cursor to no-drop. Also, changing the text to remind the user that the product is already in their cart, like in [this screenshot](https://prnt.sc/Dnoff4pV-YjQ). 

>@79ho3ein would you be willing to [open a new issue](https://github.com/woocommerce/woocommerce/issues) for this so we can add it to our tasks and prioritize? 🙇

Yes, I do that right now.

Thank you

**Reply by Joliamendi** [reply]
I completely agree with what @79ho3ein proposed. But there could be a more logical approach.

Please consider the product below.

![add to cart](https://github.com/user-attachments/assets/39c8bbd2-6d3f-498f-b5fd-a69acecfd6ad)

Wouldn't it make sense to have the 2 in stock text updated via Ajax with each click on Add to Cart? So after clicking the Add to Cart button twice, the text would change to something like "All stocks are already in your cart." (perhaps in red). In this situation, the user clearly understands why the Add to Cart button is disabled.

**Reply by Aljullu** [reply]
> Actually, by disabling the button, I meant something like reducing the opacity and changing the cursor to no-drop. Also, changing the text to remind the user that the product is already in their cart, like in [this screenshot](https://prnt.sc/Dnoff4pV-YjQ).

Yes, I would prefer something like this over completely disabling the button. :100: 

> Wouldn't it make sense to have the 2 in stock text updated via Ajax with each click on Add to Cart? So after clicking the Add to Cart button twice, the text would change to something like "All stocks are already in your cart." (perhaps in red). In this situation, the user clearly understands why the Add to Cart button is disabled.

Yes, it definitely makes sense. We are planning to make the Stock Indicator block update via the Interactivity API, but in the first iteration it will be limited to updating to display the stock status of the selected variation when attributes change in the variable product form.

But if we implement @79ho3ein's proposal, it should be possible to extend it to the Stock Indicator block and update it to show the remaining stock in real time based on the customers' products in cart.

**Comment by Aljullu** [comment]
As I know several people are following this discussion, a quick heads-up that we published a [project update in the epic](https://github.com/woocommerce/woocommerce/issues/53014#issuecomment-3008827884).

I'm also going to close this discussion as the initial decisions have been implemented. You can still leave any questions or comments in the epic: https://github.com/woocommerce/woocommerce/issues/53014.

**Comment by 79ho3ein** [comment]
Hi @Aljullu,

I wanted to share my feedback on [this PR](https://github.com/woocommerce/woocommerce/pull/59706)...

In [this video](https://github.com/woocommerce/woocommerce/pull/59706#pullrequestreview-3032246889) we see that after selecting the variables, if the product is out of stock, an out of stock label is added. Honestly, I think this is not good in terms of user experience.

### Why?

1- Because when the user changes variables, they are naturally looking at the price that is updated by selecting the variable. When the out of stock label is added, no one may notice it.

2- It doesn't make sense to display the price of a product that doesn't exist. You walk into a store and ask for the price of a product. The salesperson replies that all of those products are sold out. Would you then ask the salesperson the price of that product? Of course not. Because that product doesn't exist and in that case its price doesn't matter to you.

### What is the proposed solution?

If the product is out of stock, the out of stock label will be displayed instead of the price of that product. I think this would be much more user-friendly.

Thank you

**Reply by Aljullu** [reply]
Thanks for your feedback, @79ho3ein! We discussed it internally with designers, but we decided to keep this behavior as is, at least for now. There are several reasons a merchant might want the variation price to be visible even if the variation is out of stock, like the variation being on backorder or to let shoppers compare prices between variations.

About discoverability, the _Stock Indicator_ is added between the variation selector and the Add to Cart Button, so we expect most shoppers to notice it. That's in addition to the Add to Cart Button looking disabled in out-of-stock variations.

All of them being blocks means the _Stock Indicator_ can be moved somewhere else if desired. For example, next to the product price.

It's also possible to create new blocks when custom behaviors are needed. And if creating a new block feels overwhelming, the _Product Price_ block uses the variations' `price_html`, which can also be modified by extensions or custom code.

Hope one of those solutions works for your usecase! :slightly_smiling_face:

**Comment by 79ho3ein** [comment]
Hi @Aljullu,

I had two questions that I hope I asked in the right place.

1- Has the new "Add to cart" reached a stable state or is it clear when it will reach a stable state?

2- Is it currently possible to use "add to cart" blocks in different places? Or do they all have to be used in one parent block? For example, can we set the variable selector in a section, and the add to cart button to float (especially on mobile)?

Thank you

**Reply by Aljullu** [reply]
> I had two questions that I hope I asked in the right place.

Sure thing, and thanks for the interest!

> 1- Has the new "Add to cart" reached a stable state or is it clear when it will reach a stable state?

We don't have a target date when we plan to mark the block as stable yet. We want to be cautious before doing so, because once the block is marked as stable, there are fewer changes we can introduce. That doesn't mean the block is half-baked, though. Feature-wise everything we wanted for the first iteration is implemented.

We are currently working on:

1. Enhancing the internals, specially related to Interactivity API stores.
2. Brainstorming how extensibility will be implemented.

This work might involve making some breaking changes, that's why we keep the block marked as beta.

I would encourage you to try it out in actively maintained sites if you don't depend on any unsupported extensions. But if you don't maintain those sites actively or can't react to future breaking changes, I would stick to the classic non-blockified _Add to Cart with Options_ block for now.

> 2- Is it currently possible to use "add to cart" blocks in different places? Or do they all have to be used in one parent block? For example, can we set the variable selector in a section, and the add to cart button to float (especially on mobile)?

Right now, that's not possible because all blocks are tightly coupled to each other and to the wrapper _Add to Cart + Options_ block. However, as part of the improvements we are doing to the Interactivity API stores and extensibility, we would like each of the inner blocks to be more isolated, so they could theoretically be used on their own without the _Add to Cart + Options_ context. But we aren't there yet.

**Reply by helgatheviking** [reply]
>2- Is it currently possible to use "add to cart" blocks in different places? Or do they all have to be used in one parent block? For example, can we set the variable selector in a section, and the add to cart button to float (especially on mobile)?

Great question! I have been wondering that myself as I will also at some point want to have options separate from the add to cart button.
